src/app/plugins/scores.plugin.ts

```diff
@@ -16,294 +27,272 @@ export class ScoresPlugin extends Plugin {
-      const homeName = this._getTeamNameFromTeamData(homeData);
+    return targetGame;
+  }
 
-      matchups.push(`${visitorName}@${homeName}`);
+  private async _parseGames(message: IMessage, sport: string): Promise<Game[]> {
+    const data: string = await this._getResponse(message, sport);
+    if (!data) {
+      return [new Game()];
     }
 
-    let numFields = 0;
-    let embed = this._getNewEmbed();
-
-    //If use types 'list phone' then show the phone version
-    //This will technically work if the user types anything after 'list'
-    if (teamArg.split(' ').length == 1) {
-      matchups.forEach((matchup) => {
-        if (numFields == 25) {
-          numFields = 0;
-          embedBuffer.push(embed);
-          embed = this._getNewEmbed();
-        }
-
-        const [visitor, home] = matchup.split('@');
-        embed.addField(visitor, home, true);
-        numFields++;
-      });
-
-      //Push last embed
-      embedBuffer.push(embed);
-    } else {
-      embed.setDescription(matchups.join('\n').replace(/@/g, ' at '));
-      message.reply(embed);
+    const games: Game[] = data
+      .split(this._GAME_DELIMITER)
+      .slice(1) //First element is useless
+      .reduce((acc: Game[], gameData: string) => {
+        acc.push(this._resolveToGame(gameData));
+        return acc;
+      }, []);
+    return games;
+  }
+
+  private async _getResponse(message: IMessage, sport: string): Promise<string> {
+    const url = this._ENDPOINTS.get(sport);
+    if (!url) {
+      message.reply('Unable to locate sport.');
+      return '';
     }
-    embedBuffer.forEach((e) => {
-      message.channel.send(e);
-    });
+    return (await this.container.httpService.get(url)).data;
   }
-  private _getNewEmbed() {
-    const embed = new RichEmbed();
-    embed.setTitle('Current games available');
-    embed.setColor('#7289da');
-    return embed;
+
+  private _resolveToGame(data: string): Game {
+    let game;
+    if (data.includes('CANCELLED') || data.includes('DELAYED')) {
+      game = this._parseCancelledGame(data);
+    }
+    if (data.includes('%20at%20')) {
+      game = this._parseUpcomingGame(data);
+    }
+    if (data.includes('FINAL')) {
+      game = this._parseFinishedGame(data);
+    }
+    if (data.includes('%20IN%20') || data.includes('HALFTIME')) {
+      game = this._parseInProgressGame(data);
+    }
+
+    if (!game) {
+      game = new Game();
+    }
+
+    //For some reason a few teams names' are seperated by a different delimiter
+    game.visitorName = game.visitorName.replace(/%2[0-9]{1}/g, ' ');
+    game.homeName = game.homeName.replace(/%2[0-9]{1}/g, ' ');
+
+    return game;
   }
 
-  private _evaluateScores(a: number, b: number): number {
-    if (a == b) return 0; //Tied
-    return a > b ? 1 : -1;
+  private _parseInProgressGame(data: string): Game {
+    const [gameNumber, gameInfo] = data.split('=');
+    const [visitorData, homeInfo] = gameInfo.split('%20%20%20');
```

Comment:
consider doing a URI decode so that you're not dealing with all the URI codes

Reply:
Wow, I had no idea where the %20's and stuff were coming from. 😆 Thank you so much